src/structures/GuildMember.js

```diff
@@ -77,10 +71,19 @@ class GuildMember extends Base {
 
   _clone() {
     const clone = super._clone();
-    clone.roles = this.roles.clone();
+    clone._roles = this._roles;
```

Comment:
You should assign `this._roles.slice()` here to copy the array instead of referencing it.